deployments/azure/templates/bicep/all-in-one/all-in-one-with-alb.bicep

```diff
@@ -0,0 +1,328 @@
+targetScope = 'subscription'
+
+// default input values that should be defined at the start
+@description('name for the resource group.')
+param ResourceGroupName string = 'ai-unlimited'
+
+@description('Name for the Workspace service\'s virtual machine.')
+param AiUnlimitedName string = 'ai-service'
+
+@description('Name of the network to run the AI Unlimited service in')
+param VirtualNetworkName string = 'ai-unlimited'
+
+@description('Name of the subnet to run the AI Unlimited service in')
+param SubnetName string = 'ai-unlimited'
+
+@description('Name of the subnet for ALB to use')
+param AlbSubnetName string = 'ai-unlimited-gtw-subnet'
+
+@description('Name of the network security group')
+param SecurityGroup string = 'ai-unlimited-security-group'
+
+@description('should we create a new Azure Key Vault for bootstrapping the AI Unlimited Engine nodes.')
+@allowed(['New', 'Existing', 'None'])
+param UseKeyVault string = 'New'
+
+@description('should we create a new Azure Key Vault for bootstrapping the AI Unlimited Engine nodes.')
+param KeyVaultName string = 'ai-unlimited-kv'
+
+// inputs that are required from user
+@description('SSH public key value')
+@secure()
+param PublicKey string
+
+@description('The CIDR ranges that can be used to communicate with the AI Unlimited service instance.')
+param AccessCIDRs array = [ '0.0.0.0/0' ]
+
+@description('GUID of the AI Unlimited Role')
+param RoleDefinitionId string
+
+@description('allow access the AI Unlimited ssh port from the access cidr.')
+param AllowPublicSSH bool = false
+
+@description('allow access the AI Unlimited ssh port from the access cidr.')
+param UseSelfSignedCertForALB bool = true
+
+@description('base64 content of SSL certificate file for ALB to use, ignored if UseSelfSignedCertForALB is enabled')
+@secure()
+param ApplicationgateWayCertificate string = ''
+
+@description('password (if present) for certificate, ignored if UseSelfSignedCertForALB is enabled')
+@secure()
+param ApplicationgateWayCertificatePassword string = ''
+
+@description('Dns label prefix for ALB')
+param dnsLabelPrefix string
+
+@description('Join token for the Jupyter Labs service')
+param JupyterToken string
```

Comment:
For a better user experience, I think having a default for the JupyterToken would be helpful. 

On the develop's version, the default is set to `[uniqueString(subscription().id, utcNow())]`: https://github.com/Teradata/ai-unlimited/blob/7b52ba8c27c6b92c5249ac447c94819b606c584e/deployments/azure/bicep/all-in-one.bicep#L78